src/basic/missing_fs.h

```diff
@@ -116,3 +116,59 @@ assert_cc(FS_KEY_DESCRIPTOR_SIZE == 8);
 #ifndef FILEID_KERNFS
 #define FILEID_KERNFS 0xfe
 #endif
+
+/* linux/fsverity.h */
+#ifndef FS_IOC_ENABLE_VERITY
+/* Introduced in 085771ec14b9 ("fs-verity: add UAPI header") */
+#define FS_IOC_ENABLE_VERITY     _IOW('f', 133, struct fsverity_enable_arg)
+struct fsverity_enable_arg {
+        __u32 version;
+        __u32 hash_algorithm;
+        __u32 block_size;
+        __u32 salt_size;
+        __u64 salt_ptr;
+        __u32 sig_size;
+        __u32 __reserved1;
+        __u64 sig_ptr;
+        __u64 __reserved2[11];
+};
+#endif
+
+#if !HAVE_STRUCT_FSVERITY_DESCRIPTOR
+/* This was exposed on its own in
+ *   bde493349025 ("fs-verity: move structs needed for file signing to UAPI header")
+ */
+struct fsverity_descriptor {
+        __u8 version;           /* must be 1 */
+        __u8 hash_algorithm;    /* Merkle tree hash algorithm */
+        __u8 log_blocksize;     /* log2 of size of data and tree blocks */
+        __u8 salt_size;         /* size of salt in bytes; 0 if none */
+        __le32 __reserved_0x04; /* must be 0 */
+        __le64 data_size;       /* size of file the Merkle tree is built over */
+        __u8 root_hash[64];     /* Merkle tree root hash */
+        __u8 salt[32];          /* salt prepended to each hashed block */
+        __u8 __reserved[144];   /* must be 0's */
+};
+#endif
+
+#ifndef FS_IOC_READ_VERITY_METADATA
+/* All introduced as part of the same patch series:
+ *   e17fe6579de0 ("fs-verity: add FS_IOC_READ_VERITY_METADATA ioctl")
+ *   ...
+ *   07c99001312c ("fs-verity: support reading signature with ioctl")
+ */
+#define FS_IOC_MEASURE_VERITY   _IOWR('f', 134, struct fsverity_digest)
```

Comment:
This seems unused, and refers to `struct fsverity_digest` which isn't defined here?

Reply:
Good catch.  Will fix.

Reply:
Fixed, thanks.